plugin/nginx_plus_collectd.py

```diff
@@ -293,6 +300,21 @@ def configure(self, conf):
                 username = node.values[0]
             elif node.key == PASSWORD:
                 password = node.values[0]
+            elif node.key == API_VERSION:
+                err_msg = "{err}, please provide a valid positive integer value for the APIVersion"
+                try:
+                    api_version = int(node.values[0])
+
+                    if api_version < 0:
+                        raise ValueError("Invalid value found")
+                except ValueError as ve:
+                    raise ValueError(err_msg.format(err=ve))
+                except TypeError as te:
+                    raise TypeError(err_msg.format(err=te))
+                except Exception as e:
+                    raise Exception(err_msg.format(err=e))
```

Comment:
```suggestion
                    if api_version < 0:
                        raise ValueError("Invalid value found: {0}".format(api_version))
                except Exception as e:
                    raise type(e)(err_msg.format(err=ve))
```

Reply:
In the last pushed commit, we have made the change. Now, we are returning the caught type of error with the updated error message.